ui/app/templates/allocations/allocation/task/index.hbs

```diff
@@ -163,10 +163,10 @@
                 {{#if row.model.isCSI}}
                   <LinkTo
                     @route="csi.volumes.volume"
-                    @model={{concat row.model.volume "@" row.model.namespace.id
+                    @model={{concat row.model.source (if row.model.volumeDeclaration.perAlloc this.model.allocation.volumeExtension) "@" row.model.namespace.id
                     }}
                   >
-                    {{row.model.volume}}
+                    {{row.model.source}}
```

Comment:
The link text won't match the ID in the case of per-alloc volumes. For example, the second link here for the per-alloc volume is for `/ui/csi/volumes/csi-volume-nfs%5B0%5D@prod` so I'd expect the link text to be `csi-volume-nfs[0]`

<img width="303" alt="Screen Shot 2022-04-20 at 9 37 04 AM" src="https://user-images.githubusercontent.com/1409219/164243333-5854a71b-d913-4dee-9f16-c583c95daba5.png">

I see we also have this for `ui/app/templates/components/task-row.hbs` below, but I also can't figure out which of these two divs I'm seeing when I look at an allocation detail page like `/ui/allocations/039a9058-6e9c-a793-f589-27552d2e56c2` 😊

Reply:
Probably unrelated to this PR but while trying to get to this page from the Job page I'm sometimes getting an error with the following message:

> Error: Assertion Failed: You made a 'findRecord' request for a 'job' with id '["httpd","prod"]', but the adapter's response did not have any data

<img width="698" alt="Screen Shot 2022-04-20 at 9 43 20 AM" src="https://user-images.githubusercontent.com/1409219/164244378-bed62cc1-7513-429c-b942-abea1d76538b.png">

Reply:
> Probably unrelated to this PR but while trying to get to this page from the Job page I'm sometimes getting an error with the following message:

I've been noticing this quite a bit, too. I don't think this is newly introduced, but I'm keeping an eye on it.

Reply:
I noticed these intermittent issues as well. I think it may be related to https://github.com/hashicorp/nomad/pull/12082 but I need to investigate deeper.

Reply:
The findRecord not-founds are being tracked here: https://github.com/hashicorp/nomad/issues/12724